src/NuGet.Clients/NuGet.VisualStudio.Implementation/SolutionExplorer/Models/AssetsFileTarget.cs

```diff
@@ -0,0 +1,192 @@
+
+                if (!_dependenciesByNameAndVersion.TryGetValue(key, out dependencies))
+                {
+                    ImmutableArray<AssetsFileTargetLibrary>.Builder builder = ImmutableArray.CreateBuilder<AssetsFileTargetLibrary>(library.Dependencies.Length);
+
+                    foreach (string dependencyName in library.Dependencies)
+                    {
+                        // That there are rare cases where an advertised library is not detailed in the assets file.
+                        // For example "NETStandard.Library" as a dependency of a package brought in via a project will
+                        // not cause details NETStandard.Library to be included in the grandparent's assets file.
+                        // Such libraries are excluded.
+                        if (LibraryByName.TryGetValue(dependencyName, out AssetsFileTargetLibrary dependency))
+                        {
+                            builder.Add(dependency);
+                        }
+                    }
+
+                    dependencies = builder.Count != library.Dependencies.Length
+                        ? builder.ToImmutable()
+                        : builder.MoveToImmutable();
+                    _dependenciesByNameAndVersion[key] = dependencies;
+                }
+            }
+
+            return true;
+        }
+
+        public bool TryGetPackage(string packageId, string version, [NotNullWhen(returnValue: true)] out AssetsFileTargetLibrary? assetsFileLibrary)
+        {
+            Requires.NotNull(packageId, nameof(packageId));
+            Requires.NotNull(version, nameof(version));
+
+            if (LibraryByName.TryGetValue(packageId, out assetsFileLibrary) &&
+                assetsFileLibrary.Type == AssetsFileLibraryType.Package &&
+                assetsFileLibrary.Version == version)
+            {
+                return true;
+            }
+
+            assetsFileLibrary = null;
+            return false;
+        }
+
+        public bool TryGetProject(string projectId, [NotNullWhen(returnValue: true)] out AssetsFileTargetLibrary? assetsFileLibrary)
+        {
+            Requires.NotNull(projectId, nameof(projectId));
+
+            if (LibraryByName.TryGetValue(projectId, out assetsFileLibrary) &&
+                assetsFileLibrary.Type == AssetsFileLibraryType.Project)
+            {
+                return true;
+            }
+
+            assetsFileLibrary = null;
+            return false;
+        }
+
+        public bool TryResolvePackagePath(string packageId, string version, out string? fullPath)
+        {
+            return _snapshot.TryResolvePackagePath(packageId, version, out fullPath);
+        }
+
+        public override string ToString()
+        {
+            var s = new StringBuilder();
+            s.Append("Target \"").Append(TargetFrameworkMoniker).Append("\" ");
+            s.Append(LibraryByName.Count).Append(LibraryByName.Count == 1 ? " library" : " libraries");
```

Comment:
Localization?

Reply:
This is only for debugging. No customer would ever see this string.